src/adapter/src/session.rs

```diff
@@ -342,41 +342,51 @@ impl<T: TimestampManipulation> Session<T> {
         // This method is cancel safe because recv is cancel safe.
         loop {
             let notice = self.notices_rx.recv().await.unwrap();
-            // Filter out notices for other clusters.
-            if let AdapterNotice::ClusterReplicaStatusChanged { cluster, .. } = &notice {
-                if cluster != self.vars.cluster() {
-                    continue;
-                }
-            }
-            // Suppress replica NotReady notices until they don't occur on
-            // create/drop replica, as that leads to user confusion.
-            //
-            // TODO: Removing this variant completely, since it's the only call
-            // to broadcast_notice, causes Rust to warn about various unused
-            // functions. If we don't find another usage of notices, consider
-            // removing the entire notice system.
-            if matches!(notice, AdapterNotice::ClusterReplicaStatusChanged { .. }) {
-                continue;
-            }
-            // De-duplicate identical notices. This routinely happens for
-            // ClusterReplicaStatusChanged messages.
-            if Some(&notice) == self.prev_notice.as_ref() {
-                continue;
+            match self.notice_filter(notice) {
+                Some(notice) => return notice,
+                None => continue,
             }
-            self.prev_notice = Some(notice.clone());
-            return notice;
         }
     }
 
     /// Returns a draining iterator over the notices attached to the session.
     pub fn drain_notices(&mut self) -> Vec<AdapterNotice> {
         let mut notices = Vec::new();
         while let Ok(notice) = self.notices_rx.try_recv() {
-            notices.push(notice);
+            if let Some(notice) = self.notice_filter(notice) {
+                notices.push(notice);
+            }
         }
         notices
     }
 
+    /// Returns Some if the notice should be reported, otherwise None.
+    fn notice_filter(&mut self, notice: AdapterNotice) -> Option<AdapterNotice> {
+        // Filter out notices for other clusters.
+        if let AdapterNotice::ClusterReplicaStatusChanged { cluster, .. } = &notice {
+            if cluster != self.vars.cluster() {
+                return None;
+            }
+        }
+        // Suppress replica NotReady notices until they don't occur on
+        // create/drop replica, as that leads to user confusion.
+        //
+        // TODO: Removing this variant completely, since it's the only call
+        // to broadcast_notice, causes Rust to warn about various unused
+        // functions. If we don't find another usage of notices, consider
+        // removing the entire notice system.
```

Comment:
Is this TODO still correct? We do now have other usage of notices with @chaas's work to notify about dropping and setting clusters.

Reply:
Erp, removed this TODO and removed the code that was suppressing the notice.

Reply:
makes more sense 😅